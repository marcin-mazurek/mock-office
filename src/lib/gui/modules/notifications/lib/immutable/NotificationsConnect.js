import { connect } from 'react-redux';
import Notifications from '../Notifications';
import { newNotificationDisplayedAction, notificationsClickedAction } from '../actions';
import { allNotificationsSelector } from './selectors';

const mapStateToProps = state => ({
  notifications: allNotificationsSelector(state)
});
const mapDispatchToProps = {
  onNotificationClick: notificationsClickedAction,
  onNewNotificationDisplayed: newNotificationDisplayedAction
};
export default connect(mapStateToProps, mapDispatchToProps)(Notifications);
