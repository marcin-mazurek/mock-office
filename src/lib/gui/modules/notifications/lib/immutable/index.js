export * from './selectors';
export { default as NotificationsConnect } from './NotificationsConnect';
