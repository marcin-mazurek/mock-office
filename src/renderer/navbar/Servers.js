import React from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { getServerList } from '../servers/selectors';
import { select as dispatchSelect } from '../servers/actions';

export const Servers = ({ servers, select, goToServerPage }) => (
  <ul className="servers-list">
    {
      servers.map(server => (
        <li className="servers-list__item" key={server.id}>
          <a
            className="server-list__link"
            href=""
            onClick={(e) => {
              e.preventDefault();
              select(server.id);
              goToServerPage();
            }}
          >
            {server.name}
          </a>
        </li>
      ))
    }
  </ul>
);

Servers.propTypes = {
  servers: React.PropTypes.shape().isRequired,
  select: React.PropTypes.func.isRequired,
  goToServerPage: React.PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  servers: getServerList(state)
});

const mapDispatchToProps = {
  select: id => dispatchSelect(id),
  goToServerPage: () => push('/')
};

export default connect(mapStateToProps, mapDispatchToProps)(Servers);
