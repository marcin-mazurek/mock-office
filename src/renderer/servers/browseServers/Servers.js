import React from 'react';
import ServerListConnect from './ServerList';

const Servers = () => (
  <div>
    <header className="mdl-layout__header">
      <div className="mdl-layout__header-row">
        <span className="mdl-layout-title">
          List of servers:
        </span>
      </div>
    </header>
    <ServerListConnect />
  </div>
);

export default Servers;
