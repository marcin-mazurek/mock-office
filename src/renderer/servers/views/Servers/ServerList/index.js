import React from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { List, ListItem } from 'react-mdl';
import { getServerList } from '../../../selectors';
import { select as dispatchSelect } from '../../../actions';

const ServerList = ({ servers, select, goToServerPage }) => (
  <List>
    {
      servers.map(server => (
        <ListItem key={server.id}>
          <a
            className="server-list__link"
            href=""
            onClick={(e) => {
              e.preventDefault();
              select(server.id);
              goToServerPage();
            }}
          >
            {server.name}
          </a>
        </ListItem>
      ))
    }
  </List>
);

ServerList.propTypes = {
  servers: React.PropTypes.shape().isRequired,
  select: React.PropTypes.func.isRequired,
  goToServerPage: React.PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  servers: getServerList(state)
});

const mapDispatchToProps = {
  select: id => dispatchSelect(id),
  goToServerPage: () => push('/server')
};

export default connect(mapStateToProps, mapDispatchToProps)(ServerList);
