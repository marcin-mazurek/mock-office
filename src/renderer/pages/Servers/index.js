import React from 'react';
import ServerList from './ServerList';

const Servers = () => (
  <div>
    <header className="mdl-layout__header">
      <div className="mdl-layout__header-row">
        <span className="mdl-layout-title">
          List of servers:
        </span>
      </div>
    </header>
    <ServerList />
  </div>
);

export default Servers;
