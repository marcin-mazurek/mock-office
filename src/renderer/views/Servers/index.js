import React from 'react';
import ServerList from './ServerList';

const Servers = () => (
  <div>
    <div>List of servers:</div>
    <ServerList />
  </div>
);

export default Servers;
