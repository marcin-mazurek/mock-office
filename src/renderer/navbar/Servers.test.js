import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { List } from 'immutable';
import { Servers } from './Servers';

test('Servers should render', () => {
  const wrapper = shallow(
    <Servers
      servers={new List()}
      select={() => {}}
      goToServerPage={() => {}}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
