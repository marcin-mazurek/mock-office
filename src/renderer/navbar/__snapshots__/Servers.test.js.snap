exports[`test Servers should render 1`] = `
<ul
  className="servers-list" />
`;
